### DISCUSSION TITLE
[Doctrine\DBAL\Exception\ConnectionException (2002)] An exception occurred in the driver: SQLSTATE[HY000] [2002] Connection refused

### DISCUSSION THREAD

**Discussion by paraskumar12011996** [ddev/ddev#6412] [Q&A]
I am running my shopware 6 project through DDEV, shop storefront and admin is working fine, but db URL: 127.0.0.1:32805 is shown **127.0.0.1 sent an invalid response**. **ERR_INVALID_HTTP_RESPONSE**

When I am execute bin/build-storefront.sh command getting this below error
![image](https://github.com/user-attachments/assets/b0cc2758-c634-4829-b8a9-ef2c7a2d300c)

Through this command mysql -u db -p -h db I am getting ERROR 2003 (HY000): Can't connect to MySQL server on 'db:3306' (111)
And able to login through ddev mysql only.

So not getting the idea why this error is appending. I am using DDEV 1.23.3

**Comment by rfay** [comment]
Could you please start by doing a simple Shopware 6 quickstart, and see how that goes? https://ddev.readthedocs.io/en/stable/users/quickstart/#shopware

It sure looks like you have your database connection set up wrong. The hostname, username, database, and password should all be `db`

**Comment by stasadev** [comment]
I'm not sure what's going on here, but it's important to execute `bin/build-storefront.sh` inside the container, i.e. run it after `ddev ssh`.

And it can be related to:

- https://github.com/ddev/ddev/issues/5437#issuecomment-1905755834

These scripts may not read `.env.local` correctly.
